swf/src/avm2/read.rs

```diff
@@ -67,8 +67,10 @@ impl<'a> Reader<'a> {
 
         let len = self.read_u30()?;
         let mut method_bodies = Vec::with_capacity(len as usize);
-        for _ in 0..len {
-            method_bodies.push(self.read_method_body()?);
+        for body_idx in 0..len {
+            let body = self.read_method_body()?;
+            methods[body.method.0 as usize].body = Some(Index::new(body_idx));
```

Comment:
If FP verifyerrors here when there is already a Method attached to the MethodBody can we check and panic?

Reply:
Oh, this is in `swf`, so can't really log from there either...
And we ideally shouldn't panic in `swf` as it's more of a library :/
Gonna `Err` then.